server/src/main/java/org/apache/iotdb/db/query/externalsort/SimpleExternalSortEngine.java

```diff
@@ -0,0 +1,160 @@
+ /**
+ * Copyright © 2019 Apache IoTDB(incubating) (dev@iotdb.apache.org)
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ package org.apache.iotdb.db.query.externalsort;
+
+ import java.io.File;
+ import java.io.IOException;
+ import java.util.ArrayList;
+ import java.util.List;
+ import org.apache.commons.io.FileUtils;
+ import org.apache.iotdb.db.conf.IoTDBDescriptor;
+ import org.apache.iotdb.db.exception.StorageEngineFailureException;
+ import org.apache.iotdb.db.query.externalsort.adapter.ByTimestampReaderAdapter;
+ import org.apache.iotdb.db.query.reader.IPointReader;
+ import org.apache.iotdb.db.query.reader.IReaderByTimestamp;
+ import org.apache.iotdb.db.query.reader.chunkRelated.ChunkReaderWrap;
+
+
+ public class SimpleExternalSortEngine implements ExternalSortJobEngine {
+
+   private ExternalSortJobScheduler scheduler;
+
+   private String queryDir;
+   private int minExternalSortSourceCount;
+
+   private SimpleExternalSortEngine() {
+     queryDir = IoTDBDescriptor.getInstance().getConfig().getQueryDir();
+     minExternalSortSourceCount = IoTDBDescriptor.getInstance().getConfig()
+         .getExternalSortThreshold();
+     scheduler = ExternalSortJobScheduler.getInstance();
+
+     // create queryDir
+     try {
+       FileUtils.forceMkdir(new File(queryDir));
+     } catch (IOException e) {
+       throw new StorageEngineFailureException("create system directory failed!");
+     }
+   }
+
+   // This class is used in test.
+   public SimpleExternalSortEngine(String queryDir, int minExternalSortSourceCount) {
```

Comment:
To test this class, I suggest using
```
IoTDBDescriptor.getInstance().getConfig().setQueryDir(baseDir + "/");
IoTDBDescriptor.getInstance().getConfig().setExternalSortThreshold(2);
SimpleExternalSortEngine engine = SimpleExternalSortEngine.getInstance();
```
instead of a useless constructor in the src code.

Reply:
fixed